src/ring_theory/witt_vector/identities.lean

```diff
@@ -65,6 +65,21 @@ begin
       { exact ne_of_apply_ne (λ (j : ℕ), j.succ) hj } } }
 end
 
+variables (p R)
+
+lemma p_nonzero [nontrivial R] [char_p R p] : (p : 𝕎 R) ≠ 0 :=
```

Comment:
Does this work without the characteristic p assumption?  (Genuine question, I don't really know what to expect of the p-typical Witt vectors over a non-char-p ring.)

Reply:
I guess it should be provable, since otherwise `𝕎 R` wouldn't have characteristic 0. But "`char_zero (𝕎 R)` " isn't in the library AFAICT and I don't know how to prove it otherwise.

Reply:
Is it always of characteristic zero? I mean, even for crazy non reduced `R` or stuff like that? In any case it doesn't matter too much.

Reply:
I'm very far from an expert here, so don't take anything I say for granted! A minute on Google makes me think my assumption was wrong, you need stronger conditions than I thought for char 0. (Maybe, indeed, you need `R` to be char `p`.)